### PR TITLE
Display username on nav bar

### PR DESCRIPTION
**Pull request by anubhabsen**
On removing the import \* from nav bar file and adding a console.log in the authPage file gives the username in the console.

### CODE REVIEW

**Comment by meghprkh** [comment]
Okay, next time when you work from a stale branch, please do a:

```
git fetch --all
git reset --hard origin/master
```

More: http://stackoverflow.com/questions/1125968/force-git-to-overwrite-local-files-on-pull

And a little on history rewriting just for info: https://www.atlassian.com/git/tutorials/rewriting-history/

**Comment by meghprkh** [comment]
As a rule, make sure that whatever repo you are gonna keep contributing, never modify its master branch. Always work in temporary branches. Then when you do git pull in master, your master is clean and when starting work on another task do git branch while in master. Basically:

```
$ git status
In master
$ git checkout -b task1
# Work and commit
$ git push
# Pull request from task1
# On the side while this is getting reviewed
$ git checkout master
$ git pull
$ git checkout -b task2
# Work
# Want to fix something in task1
$ git checkout task1
# Work and fix
# Done back to task2
$ git checkout task2
```

**Comment by anubhabsen** [comment]
@meghprkh , please list all your suggestions on FB here too

**Comment by meghprkh** [comment]
Basically `auth` is a module that is responsible for handling all the data related to `auth`. It is a singleton. Basically a singleton means an object (instance of an anonymous class in kinda-c++ terminology). This object has methods and data and stores/allows to access them.

So now that we get email and name in the login and register response from backend, code it such that the auth's login function be like:

``` js
login (token, name, email) {
    auth.name = name
}
```

and then access it using `auth.name` where you want to use it

File: src/js/auth.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
 var auth = {
   x: 2,
   token: localStorage.getItem('authToken'),
-  login(authToken) {
+  login(authToken, name, email) {
     // Set the authToken and reroute to index page
+    auth.name = name
```
Comment (meghprkh) [inline src/js/auth.js]: please change to `this.name`. Consistency =)

Comment (anubhabsen) [inline src/js/auth.js]: anything else?

File: src/js/navbar.js [hunk 2]
```diff
@@ -10,7 +10,10 @@ var NavBar = props => (
     <input id='bmenub' type='checkbox' class='show'/>
     <label htmlFor='bmenub' class='burger pseudo button'>&#8801;</label>
     <div class='menu'>
+      <span>
+      <a>{auth.name}</a>
```
Comment (meghprkh) [inline src/js/navbar.js]: What is this? Just do `<span>{auth.name}</span>`. Doesnt work? Then close the span before the Link to uestions at least

File: src/js/authPage.js [hunk 3]
```diff
@@ -14,18 +15,24 @@ export default class AuthPage extends Component {
...
       break;
     case 'register':
       request.post('/auth/register')
              .send({name, email, password})
              .end((err, res) => {
                if (err) this.setState({error: true});
-               else this.props.login(res.body.token)
+               else {
+                 username = res.body.name;
+                 this.props.login(res.body.token)
```
Comment (meghprkh) [inline src/js/authPage.js]: No name and email here? Will fail on trying to login. No need to get this from server also

Comment (meghprkh) [inline src/js/authPage.js]: I mean line 34 = line 23

File: src/js/authPage.js [hunk 4]
```diff
@@ -14,18 +15,24 @@ export default class AuthPage extends Component {
...
     case 'login':
       request.post('/auth/login')
-             .send({email, password})
+             .send({name, email, password})
              .end((err, res) => {
                if (err) this.setState({error: true});
-               else this.props.login(res.body.token)
+               else {
+                 username = res.body.name;
+                 this.props.login(res.body.token, username, email)
```
Comment (meghprkh) [inline src/js/authPage.js]: Make this and the above one name and dont use undeclared vars. Use `var { name } = res.body`

File: src/js/authPage.js [hunk 5]
```diff
@@ -14,18 +15,24 @@ export default class AuthPage extends Component {
     switch (type) {
     case 'login':
       request.post('/auth/login')
-             .send({email, password})
+             .send({name, email, password})
```
Comment (meghprkh) [inline src/js/authPage.js]: Nah... no name here. Remember login does not have a name

**Comment by meghprkh** [comment]
Thanks! =)

**Pull request merged**